internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -34,16 +34,90 @@
 package clustermanager
 
 import (
+	"encoding/json"
+	"os"
+	"syscall"
+	"time"
+
 	"github.com/Azure/azure-storage-fuse/v2/common"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
+	mm "github.com/Azure/azure-storage-fuse/v2/internal/dcache/metadata_manager"
 )
 
 type ClusterManagerImpl struct {
-	storageCallback dcache.StorageCallbacks
+	nodeId string
+}
+
+// It will return online MVs as per local cache copy of cluster map
+func GetActiveMVs() []dcache.MirroredVolume {
+	return clusterManagerInstance.getActiveMVs()
+}
+
+// It will return offline/down MVs as per local cache copy of cluster map
+func GetDegradedMVs() []dcache.MirroredVolume {
+	return clusterManagerInstance.getDegradedMVs()
+}
+
+// It will return all the RVs for particular mv name as per local cache copy of cluster map
+func GetRVs(mvName string) []dcache.RawVolume {
+	return clusterManagerInstance.getRVs(mvName)
+}
+
+// It will check if the given nodeId is online as per local cache copy of cluster map
+func IsOnline(nodeId string) bool {
+	return clusterManagerInstance.isOnline(nodeId)
+}
+
+// It will evaluate the lowest number of RVs for given rv Names
+func LowestNumberRV(rvNames []string) []string {
+	return clusterManagerInstance.lowestNumberRV(rvNames)
+}
+
+// It will return the IP address of the given nodeId as per local cache copy of cluster map
+func NodeIdToIP(nodeId string) string {
+	return clusterManagerInstance.nodeIdToIP(nodeId)
+}
+
+// It will return the name of RV for the given RV FSID/Blkid as per local cache copy of cluster map
+func RvIdToName(rvFsid string) string {
+	return clusterManagerInstance.rvIdToName(rvFsid)
+}
+
+// It will return the RV FSID/Blkid of the given RV name as per local cache copy of cluster map
+func RvNameToId(rvName string) string {
+	return clusterManagerInstance.rvNameToId(rvName)
+}
+
+// It will return the nodeId/node uuid of the given RV name as per local cache copy of cluster map
+func RVNameToNodeId(rvName string) string {
+	return clusterManagerInstance.rVNameToNodeId(rvName)
+}
+
+// It will return the IP address of the given RV name as per local cache copy of cluster map
+func RVNameToIp(rvName string) string {
+	return clusterManagerInstance.rVNameToIp(rvName)
+}
+
+// Update RV state to down and update MVs
+func ReportRVDown(rvName string) error {
+	return clusterManagerInstance.reportRVDown(rvName)
+}
+
+// Update RV state to offline and update MVs
+func ReportRVFull(rvName string) error {
+	return clusterManagerInstance.reportRVFull(rvName)
 }
 
-// Start implements ClusterManager.
-func (c *ClusterManagerImpl) start(*dcache.DCacheConfig, []dcache.RawVolume) error {
+// start implements ClusterManager.
+func (cmi *ClusterManagerImpl) start(dCacheConfig *dcache.DCacheConfig, rvs []dcache.RawVolume) error {
+	cmi.nodeId = rvs[0].NodeId
```

Comment:
// Make sure it look like a valid nodeid.
common.Assert(IsValidUUID(rvs[0].NodeId), "Not a valid UUID", rvs[0].NodeId)